src/client/clientmap.cpp

```diff
@@ -644,14 +528,18 @@ void ClientMap::touchMapBlocks()
 				if not seen on display
 			*/
 
-			if (!block->mesh) {
-				// Ignore if mesh doesn't exist
-				continue;
+			v3f mesh_sphere_center;
+			f32 mesh_sphere_radius;
+			if (block->mesh) {
+				mesh_sphere_center = intToFloat(block->getPosRelative(), BS)
+						+ block->mesh->getBoundingSphereCenter();
+				mesh_sphere_radius = block->mesh->getBoundingRadius();
+			}
+			else {
+				mesh_sphere_center = intToFloat(block->getPosRelative(), BS) + v3f((MAP_BLOCKSIZE * 0.5f - 0.5f) * BS);
```

Comment:
```suggestion
			} else {
				// Not every visible block has a mesh, but we still need consider them
				mesh_sphere_center = intToFloat(block->getPosRelative(), BS) + v3f((MAP_BLOCKSIZE / 2 - 0.5f) * BS);
```
right?

Reply:
This code has been reverted altogether since it was causing problems. See comments from lhofhansl above.